### ISSUE TITLE
Ask about (i) positional embedding; (ii)  [EOS] truncation during training.

### ISSUE THREAD

**Issue by keyu-wang-2002** [zjunlp/LightThinker#12]
Hi! I’ve been following LightThinker  and making some modifications. I have two questions:

(i) I noticed that you seem to have implemented rolling_rope, but the default is "false". May I ask whether the "true" track is already implemented (considering whether it corresponds only to inference or to both training and inference)? If it is implemented, did your results show that "false" performed better than "true"? If not implemented yet, I plan to implement it myself and see.

(ii) For bs17k, more than half of the data has a length exceeding 4096. The approach in the code is to truncate directly. it seems truncating [EOS] as well.  Could this introduce bias in how vanilla and LightThinker handle the closing after solving the questions? I’m also considering other suitable training datasets, etc.

BTW, I wanted to share that in LightThinker (token), using step6_comp2, I found through my experiments that step6 can actually be much much larger, and the results can even be better. e.g., based on my training and testing results, step32_comp2 outperformed step6_comp.

**Comment by keyu-wang-2002** [comment]
thanks a lot! my question are all solved!

- Regarding bs17k truncation and [EOS], your response is consistent with my observations. I make a subset of bs17k by dropping out the samples longer than 8k, and increase the training context to 8k. The results show no significant improvements.

- Regarding increasing steps, my assumption is that the lower compression frequency and later compression help preserve intermediate reasoning information more effectively, reducing information loss and reasoning drift in long-chain reasoning. As a result, the gain on MMLU is limited, while the improvement on GSM8K is much more significant.

btw, it's normal that my results show differences. During training, I split the train data and eval data at a ratio of 9:1. I found that with the original paper's settings, overfitting occurred in less than half of whole process. Therefore, I set the learning rate to 1e-5 and selected the ckpt with the smallest eval loss for testing. Anyway, on the test task, there was no significant difference between the selected ckpt and the final overtrained ckpt.

**Comment by MikeDean2367** [comment]
OK! Thanks for your insightful findings! We are also really looking forward to seeing your new work. If you run into anything else or have more questions, please feel free to let me know! :)
